includes/course-theme/class-sensei-course-theme-styles.php

```diff
@@ -0,0 +1,253 @@
+
+		if ( ! empty( $colors ) ) {
+			$vars['text-color']             = $styles['color']['text'] ?? null;
+			$vars['background-color']       = $styles['color']['background'] ?? null;
+			$vars['primary-contrast-color'] = $vars['background-color'];
+		}
+
+		$link = $styles['elements']['link']['color'] ?? null;
+
+		if ( ! empty( $link ) ) {
+			$vars['primary-color'] = $link['text'];
+		}
+
+		return $vars;
+	}
+
+	/**
+	 * Get colors set in the customizer.
+	 */
+	private static function get_customizer_colors() {
+		$colors = [
+			'sensei-course-theme-primary-color',
+			'sensei-course-theme-background-color',
+			'sensei-course-theme-foreground-color',
+		];
+		return array_map( 'get_option', $colors );
+	}
+
+	/**
+	 * Get CSS variables for the block from its style attributes.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return array
+	 */
+	private static function get_block_colors( $attributes ) {
+
+		$styles = $attributes['style'] ?? null;
+
+		if ( empty( $styles ) ) {
+			return [];
+		}
+
+		$colors = self::get_colors( $styles );
+
+		return self::format_css_variables( $colors );
+
+	}
+
+	/**
+	 * Generate CSS variables.
+	 *
+	 * @param array  $variables Key-value pair of variable names and values.
+	 * @param string $prefix    Variable prefix.
+	 */
+	private static function format_css_variables( $variables, $prefix = '' ) {
+		$css = [];
+
+		foreach ( $variables as $variable => $value ) {
+			if ( $prefix ) {
+				$variable = $prefix . $variable;
+			}
+			if ( $value ) {
+				$css[] = sprintf( '--%s: %s;', $variable, self::get_property_value( $value ) );
+			}
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Generate a style tag with the given CSS properties.
+	 *
+	 * @param string $css      CSS properties.
+	 */
+	private static function output_style( $css ) {
+		?>
+		<style>
+			body {
+			<?php echo esc_html( $css ); ?>
+			}
+		</style>
+		<?php
+	}
+
+	/**
+	 * Converts CSS Custom Property stored as
+	 * "var:preset|color|secondary" to the form
+	 * "--wp--preset--color--secondary".
+	 *
+	 * Based on \WP_Theme_JSON::get_property_value
+	 *
+	 * @param string $value CSS value.
+	 *
+	 * @return string Style property value.
+	 */
+	private static function get_property_value( $value ) {
+
+		$prefix     = 'var:';
+		$prefix_len = strlen( $prefix );
+		$token_in   = '|';
+		$token_out  = '--';
+		if ( 0 === strncmp( $value, $prefix, $prefix_len ) ) {
+			$unwrapped_name = str_replace(
+				$token_in,
+				$token_out,
+				substr( $value, $prefix_len )
+			);
+			$value          = "var(--wp--$unwrapped_name)";
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Update the style attribute on the block wrapper element.
+	 *
+	 * Based on wp_render_elements_support
+	 *
+	 * @param string $block_content Block HTML.
+	 * @param string $style         Inline styles to add.
+	 *
+	 * @return string
+	 */
+	private static function set_style_attribute( string $block_content, string $style ) {
```

Comment:
Because normal block supports only run for blocks with a render_callback, I borrowed this from core to always add the styles to the block wrapper, even for static blocks.